AmplifyPlugins/Core/AWSPluginsCore/Model/Decorator/DirectiveDecorator.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+/// Holds all of the possible operations such as `.get`, `.list`, `.sync`, `create`, `update`, `delete`, `onCreate`,
+/// `onUpdate`, `onDelete` defined, for each GraphQL operation type (query, mutation, subscription)
+enum GraphQLOperationSubType {
+    case query(GraphQLQueryType)
+    case mutation(GraphQLMutationType)
+    case subscription(GraphQLSubscriptionType)
+}
+
+/// Replaces the directive name of the GraphQL document based on Amplify GraphQL operation types such as "get", "list",
+/// "sync", "create", "update", "delete", "onCreate", "onUpdate", and "onDelete". The GraphQL name is constructed based
+/// on the data from the Model schema and the operation type.
+public struct DirectiveDecorator: SingleDirectiveGraphQLDocumentDecorator {
```

Comment:
If all it's doing is replacing the name, consider renaming this to `DirectiveNameDecorator`